src/tunnel.c

```diff
@@ -139,6 +139,10 @@ static int pppd_run(struct tunnel *tunnel)
 			args[i++] = "ipparam";
 			args[i++] = tunnel->config->pppd_ipparam;
 		}
+		if (tunnel->config->pppd_ifname) {
+			args[i++] = "ifname";
+			args[i++] = tunnel->config->pppd_ifname;
+		}
 		// Assert that we didn't use up all NULL pointers above
```

Comment:
you should add two more NULL entries in the args[] array above just in case anyone uses all the possible arguments to pppd